dali/test/python/experimental_mode/test_batch.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import nvidia.dali.experimental.dali2 as D
+import numpy as np
+from nose_utils import attr
+from nose2.tools import params
+import nvidia.dali.backend as _b
+from nose_utils import assert_raises
+import test_tensor
+
+
+def asnumpy(batch_or_tensor):
+    if isinstance(batch_or_tensor, D.Batch):
+        return [test_tensor.asnumpy(t) for t in batch_or_tensor.tensors]
+    else:
+        return test_tensor.asnumpy(batch_or_tensor)
+
+
+@params(("cpu",), ("gpu",))
+def test_batch_construction(device_type):
+    t0 = np.array([[1, 2, 3], [4, 5, 6]], dtype=np.int32)
+    t1 = np.array([[7, 8, 9], [10, 11, 12]], dtype=np.int32)
+
+    b = D.batch(
+        [
+            D.tensor(t0),
+            D.tensor(t1),
+        ],
+        device=D.Device(device_type),
+    )
+
+    assert isinstance(b, D.Batch)
+    assert np.array_equal(asnumpy(b.tensors[0]), t0)
+    assert np.array_equal(asnumpy(b.tensors[1]), t1)
+
+
+def batch_equal(a, b):
+    for x, y in zip(a, b, strict=True):
```

Comment:
`strict=True` is new from Python 3.10. It will fail for Python 3.9, which we support.